src/hardwarewallet/signer.rs

```diff
@@ -0,0 +1,50 @@
+//! HWI Signer
+//!
+//! This module contains a implementation of a Custom Signer for rust-hwi
+
+use bitcoin::{util::bip32::Fingerprint, secp256k1::{Secp256k1, All}, psbt::PartiallySignedTransaction};
+use hwi::{HWIClient, types::{HWIDevice, HWIChain}, error::Error};
+
+use crate::signer::{SignerCommon, SignerId, TransactionSigner, SignerError};
+
+
+#[derive(Debug)]
+/// Custom Signer for Hardware Wallets
+pub struct HWISigner {
+    fingerprint: Fingerprint,
+    client: HWIClient,
+}
+
+impl HWISigner {
+    /// Create a instance from the specified device and chain
+    pub fn from_device(device: &HWIDevice, chain: HWIChain) -> Result<HWISigner, Error> {
+        let client = HWIClient::get_client(device, false, chain)?;
+        Ok(HWISigner {
+            fingerprint: device.fingerprint,
+            client,
+        })
+    }
+}
+
+impl SignerCommon for HWISigner {
+    fn id(&self, _secp: &Secp256k1<All>) -> SignerId {
+        SignerId::Fingerprint(self.fingerprint)
+    }
+}
+
+impl TransactionSigner for HWISigner {
+    fn sign_transaction(
+        &self,
+        psbt: &mut PartiallySignedTransaction,
+        _secp: &Secp256k1<All>,
+    ) -> Result<(), SignerError> {
+        psbt.combine(
+            self.client
+                .sign_tx(psbt)
+                .expect("Hardware Wallet couldn't sign transaction")
```

Comment:
Maybe in this case we should return an error, not panic?

Reply:
Hm.. The thing is, the errors (exceptions) returned by `sign_tx` are all mostly of the same type, only with different messages. So making good use of them (converting them into correct error types) will be rather... difficult. The other option is to create a separate type for python exceptions, which I feel isn't really solving the problem, just moving it.

Reply:
> only with different messages

But maybe those messages are useful to the user, to understand why the hw is not signing?